src/Expr.cpp

```diff
@@ -0,0 +1,12 @@
+#include "Expr.h"
+#include "IR.h"
+
+namespace Halide {
+// FIXME: This feels like a horrible hack. FloatImm should be able
```

Comment:
Actually, a cast of a float is also entirely consistent with how we represent integer immediates of small bit-widths, and the codegen and simplification classes happily deal with those. The comment should probably say that this is weird, but it's consistent with how we represent narrow integer immediates.